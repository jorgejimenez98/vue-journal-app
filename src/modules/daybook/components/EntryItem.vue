<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: item.id } })"
  >
    <!-- Titulo -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">{{ day }}</span>
      <span class="mx-1 fs-5">{{ month }}</span>
      <span class="mx-2 fw-ligth">{{ yearDay }}</span>
    </div>

    <div class="entry-description">
      {{ shortText }}
    </div>
  </div>
</template>

<script>
import { months, days } from "../helpers/dateHelpers";

export default {
  props: {
    item: {
      type: Object,
      required: true,
    },
  },
  computed: {
    shortText() {
      return this.item.text.length > 130
        ? this.item.text.substring(0, 130) + "..."
        : this.item.text;
    },
    day() {
      const date = new Date(this.item.date);
      return date.getDate();
    },
    month() {
      const date = new Date(this.item.date);
      return months[date.getMonth()];
    },
    yearDay() {
      const date = new Date(this.item.date);
      return `${date.getFullYear()}, ${days[date.getDay()]}`;
    },
  },
};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>
