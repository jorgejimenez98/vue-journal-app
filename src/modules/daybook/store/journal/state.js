export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',// new Date().getTime(), //12312445242,
      date: new Date().toDateString(), // sab, 27, julio,
      text: 'Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO Hola MUNDO ',
      picture: null,
    },
    {
      id: '2', //new Date().getTime() + 1000, //12312445242,
      date: new Date().toDateString(), // sab, 27, julio,
      text: 'PELICULAS PELICULAS PELICULAS PELICULAS PELICULAS PELICULAS PELICULAS PELICULAS PELICULAS ',
      picture: null,
    },
    {
      id: '3', //new Date().getTime() + 2000, //12312445242,
      date: new Date().toDateString(), // sab, 27, julio,
      text: 'JUEGO DE TRONOS JUEGO DE TRONOS JUEGO DE TRONOS JUEGO DE TRONOS JUEGO DE TRONOS ',
      picture: null,
    },
  ],
});
