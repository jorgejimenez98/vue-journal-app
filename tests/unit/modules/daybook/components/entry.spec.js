import { shallowMount } from "@vue/test-utils";
import EntryItem from "@/modules/daybook/components/EntryItem";
import { journalState } from "../../../../mocks/test-journal.state";

describe("Pruebas en EntryList component", () => {
  let wrapper;
  let mockRouter;

  beforeEach(() => {
    mockRouter = {
      push: jest.fn(),
    };
    wrapper = shallowMount(EntryItem, {
      props: {
        item: journalState.entries[0],
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });
  });

  test("Debe de hacer match con el snapchot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe de redireccionar al hacer click en el entry-container", () => {
    const routerDIV = wrapper.find('[data-testID="divRouter"]');
    routerDIV.trigger("click");

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: "-MydYHMLNbG63wnW4nuS" },
    });
  });

  test("Pruebas en las props computadas", () => {
    expect(wrapper.vm.day).toBe(20);
    expect(wrapper.vm.month).toBe("Marzo");
    expect(wrapper.vm.yearDay).toBe("2022, Domingo");
  });
});
