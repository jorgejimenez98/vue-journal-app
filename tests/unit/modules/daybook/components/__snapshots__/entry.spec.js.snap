// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en EntryList component Debe de hacer match con el snapchot 1`] = `
<div data-testid="divRouter" class="entry-container mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">20</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-ligth">2022, Domingo</span></div>
  <div class="entry-description">Madre mia</div>
</div>
`;
